File: backend/ws-service/src/main/java/com/example/wsservice/chat/service/ChatRoomService.java
package com.example.wsservice.chat.service;

import com.example.wsservice.chat.client.ChatFeignClient;
import com.example.wsservice.chat.dto.ChatRoomDto;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class ChatRoomService {

    private final ChatFeignClient chatFeignClient;

    public ChatRoomDto getChatRoom(String roomId) {
        return chatFeignClient.getChatRoom(roomId);
    }

    public ChatRoomDto updateChatRoom(String roomId, ChatRoomDto chatRoomDto) {
        return chatFeignClient.updateChatRoom(roomId, chatRoomDto);
    }
}

File: backend/ws-service/src/main/kotlin/com/example/wsservice/chat/service/ChatRoomService.kt
package com.example.wsservice.chat.service

import com.example.wsservice.chat.client.ChatFeignClient
import com.example.wsservice.chat.dto.ChatRoomDto
import org.springframework.stereotype.Service

@Service
class ChatRoomService(private val chatFeignClient: ChatFeignClient) {

    fun getChatRoom(roomId: String): ChatRoomDto = chatFeignClient.getChatRoom(roomId)

    fun updateChatRoom(roomId: String, chatRoomDto: ChatRoomDto): ChatRoomDto = chatFeignClient.updateChatRoom(roomId, chatRoomDto)

    fun updateLastReadChat(roomId: String, userId: String) = chatFeignClient.updateLastReadChat(roomId, userId)
}
